### PR TITLE
logs/no_stack_in_info

### PR DESCRIPTION
**Pull request by chylli-binary**

### CODE REVIEW

**Review by shantanu-deriv — APPROVED** [review]

File: lib/Log/Any/Adapter/DERIV.pm [hunk 1]
```diff
@@ -293,11 +298,30 @@ sub log_entry {
     );
 }
 
-sub _stderr_is_tty{
+=head2 _process_data
+
+Process the data before printing out.
+
+Parameters: $self, $data
+Return: processed data
```
Comment (sasikala-binary) [inline lib/Log/Any/Adapter/DERIV.pm:306]: suggestion use standard format for pod
https://wikijs.deriv.cloud/en/Backend/guides/perl/perl-style-guide Subroutine Comments format.

File: lib/Log/Any/Adapter/DERIV.pm [hunk 2]
```diff
@@ -293,11 +298,30 @@ sub log_entry {
...
+
+=cut
+
+sub _process_data {
+    my ($self, $data) = @_;
+    $data = clone($data);
+    return $data if(numeric_level($data->{severity}) <= numeric_level('warn'));
+    # now severity > warn
+    return $data if $self->{log_level} >= numeric_level('debug');
```
Comment (maryam-fs) [inline lib/Log/Any/Adapter/DERIV.pm:315]: based on issue's info : 

> severity of logs > warning, will print stack information if asked log level is trace or debug

where did you check if the log level is `trace`?

Comment (chylli-binary) [reply lib/Log/Any/Adapter/DERIV.pm:315]: log level >= debug, it means log level is trace or debug:
```
8         trace
7         debug
6         info (inform)
5         notice
4         warning (warn)
3         error (err)
2         critical (crit, fatal)
1         alert
0         emergency

```

**Review by qiangliu-binary — APPROVED** [review]

**Comment by sasikala-binary** [comment]
**checks tab** doesn't showing any test result. Do we need to set up any?

image

**Review by sasikala-binary — APPROVED** [review]

**Pull request merged**